tower-layer/src/util/mod.rs

```diff
@@ -47,10 +50,22 @@ where
 {
     type Response = S::Response;
     type Error = S::Error;
-    type LayerError = ();
+    type LayerError = Never;
     type Service = S;
 
     fn layer(&self, inner: S) -> Result<Self::Service, Self::LayerError> {
         Ok(inner)
     }
 }
+
+#[derive(Debug)]
+/// An error that can never occur.
+pub enum Never {}
```

Comment:
This probably should be a hidden type: public in a private mod.

Reply:
I actually had this here so that other crates can use it, is that a bad idea?

Reply:
because like 50% of layers dont produce an error so this would be useful for them to not have to always define a never type.

Reply:
What do you think @seanmonstar?